### PR TITLE
STN-362: Card Image Utilities

### PR DESCRIPTION
**Pull request by kaseybon**
# READY FOR REVIEW

## Summary
This PR adds a set of responsive utility classes to customize the width of images in card components. Classes can be applied to horizontal cards, vertical cards, and structured cards. Classes must be applied to the entire view (not just the image) and will follow a similar naming convention to the [Decanter Grid System](https://decanter.stanford.edu/page/layouts-grid-system/):
- `hb-card-image-4-of-12` sets the image to 1/4th the width of the card at all screen sizes
- `hb-card-image-md-4-of-12` sets the image to 1/4th the width of the card at the medium break point and larger

These classes have been documented in the H&S Pattern Documentation.

There should be no change to any existing card patterns, the change is purely additive.

## Need Review By (Date)
05/18/2020

## Urgency
medium

## Steps to Test
1. Confirm that all tests pass
2. Confirm that all card patterns have not changed in their default state
3. Modify a view and apply the following classes to the entire view:
    `hb-card-image-12-of-12 hb-card-image-sm-6-of-12 hb-card-image-md-3-of-12`
    Custom_Econ_News__Content____Department_of_Economics
    This should cause the width of the image to be 100% of the card at the smallest screen sizes then change to 50% at the small breakpoint, and then be 25% of the width at the medium breakpoint.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_card-images.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Loops through the number of available columns (12). Classes determine the number
+// of columns that the image in cards will span.
+@mixin hb-image-sizes() {
+  @for $i from 1 through 12 {
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_card-images.scss:4]: These screenshots are from http://economics.suhumsci.loc/about/news.
At `6-of-12` the content overflows the containing div of `hb-card--horizontal`.

6-of-12

<br><br>

At `7-of-12` and above the content causes horizontal scroll in Safari and is cut off in Chrome. This is at a scrren width of ~1200px.

chrome

<br><br>

`6-of-12` and above also do not work well with a 2nd sidebar. This screenshot is with an image at `7-of-12`.

with 2nd sidebar

Comment (heatherdesigns) [reply docroot/themes/humsci/humsci_basic/src/scss/utilities/_card-images.scss:4]: The vertical cards are fine. And they would benefit the most from `6-of-12` and up. Anything smaller looks awkward. I hate to say that we may need different utility classes for vertical and horizontal cards, but maybe?

**Review by heatherdesigns — CHANGES_REQUESTED** [review]
Fantastic idea creating utility classes that are similar to those used in the local footer which is something H&S is familiar with! 🏆 

When `hb-card-image-6-of-12` and above are used on horizontal and structured cards there is an overflow of content and horizontal scroll. I think the values of flex and max-width in `hb-card__content` need to be adjusted as well. When an image is using `flex-column(7)`, then the content needs `flex-column(5)` to prevent overflow and horizontal scroll.

**Comment by kaseybon** [comment]
@heatherdesigns I fixed the utility classes to also adjust the width of the content area.

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_card-images.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    &-#{$i}-of-12 {
+      .hb-card__img {
+        @include flex-column($i);
+      }
+
+
+      @include grid-media-min($hb-horizontal-card-breakpoint) {
+        .hb-card--horizontal .hb-card__content {
+          @include flex-column(12 - $i);
```
Comment (heatherdesigns) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_card-images.scss:13]: Nice!! 🏆

**Review by heatherdesigns — APPROVED** [review]

**Pull request merged**